packages/ui/src/components/DataStory/clients/SocketClient.tsx

```diff
@@ -1,155 +1,173 @@
-import { Diagram, NodeDescription } from '@data-story/core';
+import { createDataStoryId, Diagram, Hook, NodeDescription } from '@data-story/core';
 import { ServerClient } from './ServerClient';
-import { Hook } from '@data-story/core';
 import { eventManager } from '../events/eventManager';
 import { DataStoryEvents } from '../events/dataStoryEventType';
+import { catchError, filter, firstValueFrom, map, Observable, retry, timeout } from 'rxjs';
+import { webSocket, WebSocketSubject } from 'rxjs/webSocket';
+import { TableItems } from './ItemsApi';
 
 export class SocketClient implements ServerClient {
-  protected socket?: WebSocket;
+  protected socket$: WebSocketSubject<any>;
+  protected wsObservable: Observable<any>;
   protected maxReconnectTries = 100;
-  protected reconnectTimeout = 1000;
-  protected reconnectTries = 0;
-  protected itemStorage = new Map<string, any[]>();
+  protected reconnectTimeoutMs = 1000;
 
   constructor(
     protected setAvailableNodes: (nodes: NodeDescription[]) => void,
     protected updateEdgeCounts: (edgeCounts: Record<string, number>) => void,
-  ) {}
+  ) {
+    this.socket$ = webSocket({
+      url: 'ws://localhost:3100',
+      openObserver: {
+        next: () => {
+          console.log('Connected to server: localhost:3100');
+          this.describe()
+        }
+      },
+      closeObserver: {
+        next: () => {
+          console.log('WebSocket closed.');
+        }
+      }
+    });
+
+    this.wsObservable = this.socket$.pipe(
+      retry({ count: this.maxReconnectTries, delay: this.reconnectTimeoutMs }),
+    )
+  }
 
   itemsApi = () => {
     return {
-      getItems: async ({
+      getItems: ({
         atNodeId,
         limit = 10,
         offset = 0,
-      }: {
-        atNodeId: string,
-        limit?: number,
-        offset?: number,
-
-      }) => {
-        const items: any[] = this.itemStorage.get(atNodeId) || [];
+      }: TableItems) => {
+        const msgId = createDataStoryId();
+        this.socketSendMsg({
+          type: 'getItems',
+          atNodeId,
+          id: msgId,
+          limit,
+          offset,
+        });
 
-        return items.slice(offset, offset + limit);
+        return firstValueFrom(this.wsObservable.pipe(
+          filter(it => it.type === 'UpdateStorage' && it.id === msgId),
```

Comment:
Can we type "it"? And maybe rename it to "message" as to not confuse it with "items"?